src/installer/installer.go

```diff
@@ -883,8 +883,17 @@ func (i *installer) checkHostname() error {
 	}
 
 	data := fmt.Sprintf("random-hostname-%s", uuid.New().String())
-	i.log.Infof("Hostname [%s] is invalid, generated random hostname [%s] and writing data into /etc/hostname")
-	return i.ops.CreateRandomHostname(data)
+	i.log.Infof("Hostname [%s] is invalid, generated random hostname [%s] and writing data into /etc/hostname", data)
+	if err := i.ops.CreateRandomHostname(data); err != nil {
+		i.log.Errorf("Failed to generate random hostname", err)
+		return err
+	}
+
+	if highAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+		i.log.Infof("Restarting systemd-hostnamed service for SNO after writing new hostname [%s]", data)
+		i.ops.SystemctlAction("restart", "systemd-hostnamed")
```

Comment:
Wondering, maybe it is just easier to run hostnamectl command for all cases? it should set everything as expected without need to restart service?

Reply:
Good idea to use hostnamectl, will update. Do you also mean to do it on multi-node clusters too?

Reply:
I tested both ways and I get this when trying to access the installed spoke cluster 
```
Unable to connect to the server: tls: failed to verify certificate: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kube-apiserver-lb-signer")
```

Reply:
Should I send the requested hostname to the installer?

Reply:
very strange, hostname is not part of kubeconfig

Reply:
very strange, hostname is not part of kubeconfig

Reply:
Ah sorry this error was a complete fluke. Something weird with my env 😅 Made the change to use `hostnamectl` to set the hostname for any cluster (not just SNO)!